aten/src/ATen/native/im2col_shape_check.h

```diff
@@ -37,9 +37,11 @@ static inline void col2im_shape_check(
       dilation_width);
 
   int64_t ndim = input.ndimension();
+  // allow dim=0 only the batch dimension.
   TORCH_CHECK(
-      input.numel() != 0 && (ndim == 2 || ndim == 3),
-      "Expected non-empty 2D or 3D input tensor, but got input of sizes",
+      (ndim == 2 && input.size(1) != 0) ||
```

Comment:
same as in replication pad, if ndim==2 you need size(1) and size(0) both to be non-zero, similar for other checks

Reply:
Ah I see.. thanks for that I'll send a patch soon.